### DISCUSSION TITLE
Palm PDB - unable to view/convert

### DISCUSSION THREAD

**Discussion by nameauser** [apprenticeharper/DeDRM_tools#1700] [Q&A]
I read a [2018 guide](https://www.epubor.com/how-to-remove-drm-from-pdb-and-convert-pdb-to-pdf.html) to do this  which suggested a specific eReaderPDB2PML_vXX_plugin.zip within the main zip file. It is not there anymore in the current version.

Palm PDB does not appear to be a valid source in the list of formats under Configuration.

I still have the Palm (Garmin iQue3600) device, but it has since been reverted to factory condition without any of the apps I once had. I have bought all the ebooks long ago for this device but am unsure what the unlock password/code is, and the only visible device id is a serial number on a sticker on the back.

I really would like to access to those children stories as none of them could be found online anymore even though I am prepared to pay for them again in a modern ebook format.

I have the debug log below.

Would appreciate any help on this. Thanks.

[tmph1merhye.txt](https://github.com/apprenticeharper/DeDRM_tools/files/6471734/tmph1merhye.txt)

**Comment by ElleKayEm** [comment]
That post may be dated 2018, but the info is way older than that (before these tools were made into one DeDRM plugin).  The section of the current plugin that handles pdb files is "eReader ebooks" (the last place I'm aware was using that file type was Fictionwise which closed almost a decade ago).  But it needs to be configured with name and password (which for Fictionwise was the name and last 8 digits of the credit card on file at the time of download).  If you had that info, I'd stick with the last Python 2 versions (calibre 4.23 and DeDRM 6.8.1).  The current Python 3 versions probably haven't been tested for that file type.

**Reply by nameauser** [reply]
Thank you so much. I think I can recall the info. Do you know how the name and card digits are put together? Is it a string without spaces like "johndoe12345678"?

**Reply by ElleKayEm** [reply]
DeDRM's customization dialog has separate fields for name and the last 8 digits.  Enter the name as "John Doe" and the digits with no spaces.  It also asks for a Unique Key Name.  This can be whatever you want to call it and will show in the list of keys.

calibre 4.23:  https://download.calibre-ebook.com/4.23.0/
DeDRM 6.8.1:  https://github.com/apprenticeharper/DeDRM_tools/releases/tag/v6.8.1

**Reply by nameauser** [reply]
thank you ElleKayEm, installed 4.23 and 6.8.1 but still can't get the files to open or convert yet. Need to figure out key I guess

**Reply by nameauser** [reply]
I have tried but failed to unlock them. Is there no other way of accessing them? I looked at error logs and it is a key error

**Reply by ElleKayEm** [reply]
Not that I know of.  The key is a hash created from the name and cc#.

**Reply by nameauser** [reply]
thanks, still trying. Do I need to save the key as a .b63 file each time? And do I need to close and reopen after each failed attempt to convert? Also, do I need to remove the book each time?

**Reply by nameauser** [reply]
I tried to view, again DRM pop-up, but the log does not seem to show any attempt at using the key.
[tmph1merhye.txt](https://github.com/apprenticeharper/DeDRM_tools/files/6474352/tmph1merhye.txt)

**Reply by nameauser** [reply]
And conversion attempt log seems to show "DRM not supported", rather than wrong key error
[tmpizsatj.txt](https://github.com/apprenticeharper/DeDRM_tools/files/6474391/tmpizsatj.txt)

**Reply by ElleKayEm** [reply]
You would only need to save the key as a .b63 file if you were importing the key into another calibre setup.  After adding a key, click ok and apply out of the dialog boxes, restart calibre, and add your book.  DeDRM will attempt to remove the DRM when you first add the book.  The later versions of DeDRM also attempt to remove DRM when you try to convert (if necessary).  I thought that was the case with DeDRM 6.8.1 -- looks like you have 6.8.0.  You didn't trigger the DeDRM plugin in either of those logs so that I assume that error is just calibre saying it can't convert something with DRM.
